### PR TITLE
Regex, objects, bits

### PR DESCRIPTION
**Pull request by anderseknert**
Signed-off-by: Anders Eknert <anders@eknert.com>

### CODE REVIEW

File: src/test/clojure/jarl/builtins/regex_test.clj
Comment (johanfylling) [inline src/test/clojure/jarl/builtins/regex_test.clj:23]: `0` as the `number` arg could be an additional interesting edge case to test.

File: src/main/clojure/jarl/builtins/regex.clj
Comment (johanfylling) [inline src/main/clojure/jarl/builtins/regex.clj:41]: Bummer. Should be possible to find a suitable solution. A task for another day 😄 .

File: src/test/clojure/jarl/builtins/strings_test.clj
Comment (johanfylling) [inline src/test/clojure/jarl/builtins/strings_test.clj:13]: 😆

**Review by johanfylling — APPROVED** [review]
A nice set of additions to the ever growing number of supported built-ins 😃 .

File: src/main/clojure/jarl/builtins/bits.clj
Comment (anderseknert) [reply src/main/clojure/jarl/builtins/bits.clj:5]: Indeed. I think this might even have been done in the other PR, so holding off on this until that's been merged.

File: src/test/clojure/jarl/builtins/bits_test.clj
Comment (anderseknert) [reply src/test/clojure/jarl/builtins/bits_test.clj:9]: Yes, but it would not mirror what we'd see in OPA, would it? :)

File: src/main/clojure/jarl/builtins/bits.clj
Comment (anderseknert) [reply src/main/clojure/jarl/builtins/bits.clj:5]: Indeed, I've stopped adding type-checks to the builtins here as I think we'll want to have that in the evaluator instead, based on the argument types provided by the plan for each builtin. WDYT?

File: src/main/clojure/jarl/builtins/objects.clj
Comment (anderseknert) [reply src/main/clojure/jarl/builtins/objects.clj:10]: indeed :D

File: src/main/clojure/jarl/builtins/regex.clj
Comment (anderseknert) [reply src/main/clojure/jarl/builtins/regex.clj:41]: For sure! I'm still not sure why they'd do that!

File: src/test/clojure/jarl/builtins/bits_test.clj
Comment (anderseknert) [reply src/test/clojure/jarl/builtins/bits_test.clj:18]: Yeah my current thinking is that we'll remove this from each individual built-in in favor for a more general purpose check in the evaluator when calling _any_ built-in. The built-ins may still provide more granularity though, just like in OPA.. so if the evaluator checks for "number" we may still have the built-in check if it's an integer, and so on.

File: src/test/clojure/jarl/builtins/objects_test.clj
Comment (anderseknert) [reply src/test/clojure/jarl/builtins/objects_test.clj:35]: Yep, `json.remove` does this.

File: src/test/clojure/jarl/builtins/regex_test.clj
Comment (anderseknert) [reply src/test/clojure/jarl/builtins/regex_test.clj:24]: Sure! Added.

**Pull request merged**
